CDK/Linux/hello-world/lib/aws-image-builder-stack.ts

```diff
@@ -75,62 +50,114 @@ export class ImageBuilderStack extends Stack {
       );
     }
   }
+
   private createImageBuilderByConfig(
     imageBuilderPipeline: any,
     imageBuilderSubnetId: string,
     imageBuilderSG: SecurityGroup,
     vpc: IVpc
   ) {
-    const dir = imageBuilderPipeline.dir;
-    const files = readdirSync(dir);
-    if (files.some((f) => !existsSync(f))) {
-    }
-    const componentList: ImageBuilderComponent[] = files.map((file) => ({
-      name: file.split(".")[0],
-      data: this.getData(dir, file),
-    }));
-    const parentImage: ParentImage = imageBuilderPipeline.parentImage;
-
-    const amiIdLocation = new StringParameter(this, "Parameter", {
-      description: `The value of image`,
-      parameterName: `ec2image_ami_${imageBuilderPipeline.name}`,
-      stringValue: "n/a",
+    // get component list
+    const componentList = this.parseComponentList(imageBuilderPipeline.components);
+
+    const amiIdSSMParameter = new StringParameter(this, 'amiIDSSMParameter', {
+      description: `The id of the ec2 AMI image that is built by image builder pipeline ${imageBuilderPipeline.name}`,
+      parameterName: `imagebuilder_ami_${imageBuilderPipeline.name}`,
+      stringValue: 'n/a',
       tier: ParameterTier.ADVANCED,
     });
 
     new AWSImageBuilderConstruct(
       this,
       `AWS-ImageBuilder-Events-${imageBuilderPipeline.name}`,
       {
-        imageBuilderToolsBucket: this.imageBuilderToolsBucket,
         name: imageBuilderPipeline.name,
         subnetId: imageBuilderSubnetId,
         imageBuilderSG: imageBuilderSG,
         instanceProfileName: imageBuilderPipeline.instanceProfileName,
         imageBuilderComponentList: componentList,
         cfnImageRecipeName: imageBuilderPipeline.cfnImageRecipeName,
         version: imageBuilderPipeline.version,
-        parentImage: parentImage,
-        amiIdLocation: amiIdLocation,
+        parentImage: imageBuilderPipeline.parentImage,
+        amiIdSSMParameter: amiIdSSMParameter,
         vpc: vpc,
       }
     );
   }
 
-  validAndGetPipelineConfiguration() {
+  private parseComponentList(components: string[]): ImageBuilderComponent[] {
+    // The entry in the components can be in one of the following three types
+    // - arn for AWS managed components
+    // - a directory name that contains one or more component yaml files in it
+    // - a specific path to a component yaml file
+    // this function will parse the list and generate entries for each of them
+
+    const componentList: ImageBuilderComponent[] = [];
+
+    for (const component of components) {
+      // test if component is a valid AWS managed component arn
+      const arnMatch = component.match(
+        /^arn:aws:imagebuilder:[a-z0-9\-]*:aws:component\/([\w\-]*)\/.*/
```

Comment:
Given this sample could be tried in any partition, can you adjust to not use the `aws` partition directly in this regex?

Reply:
Regex is updated to accommodate all known partition values at the moment.